### PR TITLE
Update public facing map and navbar

### PR DESCRIPTION
**Pull request by juliexxia**
Remove initial dialog from public-facing map page, automatically fall back to using token server and add a snackbar while waiting for the token

Use a fetch of usershapes to determine whether to show hamburger nav menu or just 'public' title and help link. Shared promise with the usershapes fetch in polygon_draw.js. 

Along the way, update upload_image_collection.html to use our downloaded jquery lib (fixed some issues my PR created in navbar.js)

Along the way, fix a snackbar bug where classes were shared across snackbars.

Along the way, fix a bug where the page hangs forever if the user is not logged in at all (repro: open a new incognito window and visits page)

Public view: 
image

### CODE REVIEW

**Comment by juliexxia** [comment]
@RamV13 Do you think you have time to make the public view slightly prettier once this PR gets submitted? I did bare bones to make it not horrible but could definitely use your styling touch (:

File: cypress/integration/unit_tests/color_function_util_test.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ describe('Unit tests for color function utility', () => {
 
   before(() => {
     cy.visit('test_utils/empty.html');
-    return cy.document().then((doc) => {
```
Comment (janakdr) [inline cypress/integration/unit_tests/color_function_util_test.js]: nit: unrelated?

File: cypress/integration/unit_tests/navbar_test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import {loadNavbar} from '../../../docs/navbar.js';
+import * as NavbarLib from '../../../docs/navbar_lib.js';
+import {loadScriptsBeforeForUnitTests} from '../../support/script_loader';
+
+describe('Unit test for navbar', () => {
```
Comment (janakdr) [inline cypress/integration/unit_tests/navbar_test.js]: nit: can we get rid of describe since it's useless?

File: cypress/integration/unit_tests/navbar_test.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      navPublic.append(publicHelper);
+
+      const navHeader = doc.createElement('h1');
+      navHeader.id = 'nav-title-header';
+      doc.body.append(navHeader, navPublic, navToggle);
+    });
+  });
+
+  let getUrlStub;
+  beforeEach(() => {
```
Comment (janakdr) [inline cypress/integration/unit_tests/navbar_test.js]: nit: I think you can get rid of braces here and below

File: docs/authenticate.js [hunk 4]
```diff
@@ -145,32 +140,10 @@ class Authenticator {
...
-                        this.eeInitializeCallback, defaultErrorCallback));
-              },
-            },
-          ],
-          modal: true,
-          width: 600,
-          close: () => this.requireSignIn(),
+        showToastMessage('fetching ee token', -1);
+        this.getAndSetEeTokenWithErrorHandling().then(() => {
+          showToastMessage('fetching ee token', -1);
```
Comment (janakdr) [inline docs/authenticate.js]: why show it again here?

File: docs/firestore_document.js [hunk 5]
```diff
@@ -48,6 +50,18 @@ function getDisasters() {
...
+/**
+ * @return {firebase.firestore.CollectionReference} all usershapes collection
+ */
+function userFeatures() {
+  return getFirestoreRoot().collection('usershapes');
+}
+
+/** @return {Promise<firebase.firestore.QuerySnapshot>} all user shapes */
+function getUserFeatures() {
+  return userFeatures().get();
```
Comment (janakdr) [inline docs/firestore_document.js]: might as well follow the pattern of what I did in #414 and do what happens in `getDisastersData` below (can extract that code to a common function), so user gets back a map, rather than having to iterate themselves.

File: docs/navbar.js [hunk 6]
```diff
@@ -1,77 +1,92 @@
...
- * Loads the navbar and invokes the callback upon load.
- *
- * @param {Function} callback the callback invoked upon load
+ * Callback function for loading navbar.html that updates the bar based
+ * on what kind of user is looking at it (firebase privileged or not).
+ * @param {Promise<Map<string, Object>>} firebaseDataPromise Promise with data
+ *     for all disasters
+ * @param {?Function} changeDisasterHandler Function invoked when current
+ *     disaster is changed. location.reload is called if null
+ * @param {Promise<any>} privilegedUserPromise Promise with user shapes data *
```
Comment (janakdr) [inline docs/navbar.js]: nit: stray * at end of line

File: docs/navbar.js [hunk 7]
```diff
@@ -1,77 +1,92 @@
...
-    $('#manage-disaster-a')
-        .prop('href', getUrlUnderDocs(MANAGE_DISASTERS_PAGE));
-    $('#upload-image-collection-a')
-        .prop('href', getUrlUnderDocs(UPLOAD_IMAGE_COLLECTION_PAGE));
-    $('#help-a').prop('href', getHelpUrl());
-    callback();
-  });
+async function loadNavbar(
+    firebaseDataPromise, changeDisasterHandler, privilegedUserPromise, title) {
+  // console.log($('#nav-toggle'));
```
Comment (janakdr) [inline docs/navbar.js]: nit: debugging, here and below

File: docs/navbar.js [hunk 8]
```diff
@@ -1,77 +1,92 @@
...
  * Loads the navbar with a disaster picker.
- * @param {Promise} firebaseAuthPromise Promise that completes when Firebase
- *     login is done
- * @param {string} title Title of page
- * @param {?Function} changeDisasterHandler Function invoked when disaster is
- *     changed. If not specified, reloads page
- * @param {?Promise<Map<string, Object>>} firebaseDataPromise If caller has
- *     already kicked off a fetch of all disasters, pass that Promise in here to
- *     avoid a duplicate fetch
+ * @param {Object} data
```
Comment (janakdr) [inline docs/navbar.js]: helpful to use the `@param data.firebaseAuthPromise` syntax below data so arguments are documented. Can do that here rather than in body of function. I think you can also directly destructure, just declaring the function
`function loadNavbarWithPicker({firebaseAuthPromise, title...}) {`

File: docs/navbar.js [hunk 9]
```diff
@@ -1,77 +1,92 @@
...
-/**
- * Gets the url for the help section relevant to the current page
- * @return {string}
- */
-function getHelpUrl() {
-  if (window.location.pathname.endsWith(MANAGE_LAYERS_PAGE)) {
-    return MANAGE_LAYERS_HELP_URL;
-  } else if (window.location.pathname.endsWith(MANAGE_DISASTERS_PAGE)) {
-    return MANAGE_DISASTERS_HELP_URL;
+  // On every page other than the main map, this will be null but if you're
```
Comment (janakdr) [inline docs/navbar.js]: Maybe name this something more generic? Also, why not make this promise trivially resolve if caller passes in null, meaning they know it will succeed? (Or callers could pass in trivial promise themselves.)

File: docs/navbar_lib.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import {HELP_DOC_URL, MANAGE_DISASTERS_HELP_URL, MANAGE_LAYERS_HELP_URL} from './help.js';
+import {inProduction} from './in_test_util';
```
Comment (janakdr) [inline docs/navbar_lib.js]: you need the '.js' for this to work on web.

File: docs/polygon_draw.js [hunk 11]
```diff
@@ -448,10 +447,12 @@ function createHelpIcon(url) {
  * @param {google.maps.Map} map Map to display regions on
  * @param {Promise<any>} firebasePromise Promise with Firebase damage data (also
  *     implies that authentication is complete)
+ * @param {Promise<any>} userShapesPromise Promise with user shapes data
```
Comment (janakdr) [inline docs/polygon_draw.js]: nit: can type this more strictly, especially after making my proposed change to firestore_document

File: token_server/token_server.js [hunk 12]
```diff
@@ -34,7 +34,8 @@ const TIME_BEFORE_REGENERATION = 40 * ONE_MINUTE_IN_MILLISECONDS;
 // Cope with slight differences between ESM/Node transpilation of googleapis.
 const googleAuth = GoogleAuth.default || GoogleAuth;
 
-const allowedOrigins = new Set(['https://givedirectly.github.io']);
+const allowedOrigins =
+    new Set(['https://givedirectly.github.io', 'http://localhost:8080']);
```
Comment (janakdr) [inline token_server/token_server.js]: can you remove this and redeploy when done testing?

File: docs/navbar_lib.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+/**
+ * Get url of a file in or below our docs directory by using the path of this
+ * current script.
+ * @param {string} pathFragment path fragment to append to '.../docs/'
+ * @return {string}
+ */
+function getUrlUnderDocs(pathFragment) {
+  return inProduction() ? import.meta.url.replace(/navbar\.js$/, pathFragment) :
```
Comment (janakdr) [inline docs/navbar_lib.js]: this needs to be navbar_lib.js now.

File: docs/authenticate.js [hunk 14]
```diff
@@ -145,32 +140,10 @@ class Authenticator {
...
-                this.getAndSetEeTokenWithErrorHandling().then(
-                    () => initializeEE(
-                        this.eeInitializeCallback, defaultErrorCallback));
-              },
-            },
-          ],
-          modal: true,
-          width: 600,
-          close: () => this.requireSignIn(),
+        showToastMessage('fetching ee token', -1);
```
Comment (janakdr) [inline docs/authenticate.js]: Maybe expand this a bit? "Fetching EarthEngine token" at least? But maybe something like "Not whitelisted for EarthEngine access: trying anonymous access"

File: docs/navbar.html [hunk 15]
```diff
@@ -1,7 +1,12 @@
 <div class="navbar">
   <div class="nav-title"><h1 class="nav-header" id="nav-title-header"></h1></div>
   <div id="nav-left" class="nav-left"></div>
-  
+
+  <!--  TODO: style these better-->
```
Comment (RamV13) [inline docs/navbar.html]: Yup I can take this on. Please feel free to mark this TODO against me.

File: cypress/integration/unit_tests/polygon_draw_firebase_issues_test.js [hunk 16]
```diff
@@ -65,20 +67,15 @@ describe('Tests for polygon_draw.js with Firebase issues', () => {
                 'Error retrieving user-drawn features. Try refreshing page');
     const error = new Error();
     error.code = 'not a code we know';
-    const getStub = cy.stub().throws();
-    const collectionStub = cy.stub().returns({get: getStub});
     const getFirestoreStub =
-        cy.stub(FirestoreDocument, 'getFirestoreRoot').returns({
-          collection: collectionStub,
-        });
+        cy.stub(FirestoreDocument, 'userFeatures').throws(error);
```
[the remark this answers is not in the document]
Comment (janakdr) [reply cypress/integration/unit_tests/polygon_draw_firebase_issues_test.js]: Sorry, this was a TODO to myself that I forgot to delete. It was gone before you responded, but I guess github didn't show it to you.
